core/types/receipt.go

```diff
@@ -258,6 +258,56 @@ func (r *Receipt) Size() common.StorageSize {
 	return size
 }
 
+// DeriveFields fills the receipt with computed fields based on consensus
+// data and contextual infos like containing block and transactions.
+func (r *Receipt) DeriveFields(
+	signer Signer, hash common.Hash, number uint64, time uint64, baseFee *big.Int, blobGasPrice *big.Int,
+	gasUsedSoFar uint64, logIndx uint, txn *Transaction, txnIndx uint,
```

Comment:
Some parameters of this function need to be turned into a struct, like `DeriveReceiptContext`. Everything except `signer` should be in the struct. It can be passed by value to avoid allocation for the struct in `Receipts.DeriveFields`.